core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -1150,6 +1150,49 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     assert(sc.listJars().exists(_.contains("org.apache.hive_hive-storage-api-2.7.0.jar")))
     assert(sc.listJars().exists(_.contains("commons-lang_commons-lang-2.6.jar")))
   }
+
+  test("SPARK-34069: Kill barrier tasks should respect SPARK_JOB_INTERRUPT_ON_CANCEL") {
```

Comment:
Shall we move it to `BarrierTaskContextSuite`?

Reply:
Yes, that's more clearer.